skiko/src/commonMain/kotlin/org/jetbrains/skia/paragraph/TextStyle.kt

```diff
@@ -359,10 +378,31 @@ class TextStyle internal constructor(ptr: NativePointer) : Managed(ptr, _Finaliz
         return this
     }
 
+    private inline fun Float.asNumberOrNull(): Float? = if (isNaN()) null else this
+
     val fontMetrics: FontMetrics
         get() = try {
             Stats.onNativeCall()
-            _nGetFontMetrics(_ptr)
+            val fontMetricsData = withResult(FloatArray(15)) {
```

Comment:
Same as above

Reply:
Agree, generally extensions on primitives better be avoided for cases like that.

Reply:
@olonho @VeselovAlex yep, that totally make sense - I've moved all to the fromInteropPointer Companion extension functions (it seem to me that even Companion object per se should be agnostic of such  functions) - regarding the fact that this PR otherwise is approved I'll merge it after the pipeline check pass.

If you have strong arguments in favor of keeping such functions inside the Companion itself - let's discuss - if there are strong arguments I'll just will introduce this changed